parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -116,9 +120,45 @@ public void setPageSource(PageReadStore pageStore, long rowPosition) {
 
   static class ConstantReader<C> implements ParquetValueReader<C> {
     private final C constantValue;
+    private final TripleIterator<?> column;
+    private final List<TripleIterator<?>> children;
 
     ConstantReader(C constantValue) {
       this.constantValue = constantValue;
+      this.column = NullReader.NULL_COLUMN;
+      this.children = NullReader.COLUMNS;
+    }
+
+    ConstantReader(C constantValue, int definitionLevel) {
+      this.constantValue = constantValue;
+      this.column = new TripleIterator<Object>() {
+        @Override
+        public int currentDefinitionLevel() {
+          return definitionLevel;
+        }
+
+        @Override
+        public int currentRepetitionLevel() {
+          return 0;
+        }
+
+        @Override
+        public <N> N nextNull() {
+          return null;
+        }
+
+        @Override
+        public boolean hasNext() {
+          return false;
+        }
+
+        @Override
+        public Object next() {
+          return null;
+        }
+      };
+
+      this.children = ImmutableList.of(column);
```

Comment:
Trying to understand how this works, but this previously is empty list, but we need it here?

Reply:
Never mind, I guess its like that in other readers, bit confusing.

Reply:
Yeah, this is just keeping the same as the original code.